src/solve.jl

```diff
@@ -177,7 +177,7 @@ function __init__()
 				end
 				NLopt.min_objective!(opt, fg!)
 			else 
-				_loss = function(θ)
+				_loss = function(θ,G)
```

Comment:
we need to update the gradient here.

Reply:
In the previous discussions we decided to allow gradients and hessians only with the `OptimizationFunction` interface, this branch of `else` deals with the directly creating an `OptimizationProblem` with your function so I intentionally left out the gradient update, we do that correctly in the other branch https://github.com/SciML/GalacticOptim.jl/blob/270cf27d83fa83a3ded273babe884d3ef014f79a/src/solve.jl#L168 and for every other backend

Reply:
Yes, and here is where you'd grab and use `fg!`

Reply:
And you can't just not supply the gradient here, or else it's incorrect.

Reply:
@ChrisRackauckas the point is that when `OptimizationFunction` is not used, we expect the user to not want to use a gradient (or higher derivative) based method and thus there is no need to update it here. Only when the objective is wrapped in `OptimizationFunction` we automatically generate the derivatives or allow the user to pass those in to be used in the optimiser.

Reply:
isn't there a default AD it falls back to?

Reply:
and what would you do in this case? Throw an error before the optimization? Since NLopt will expect `G` to be updated.

Reply:
Yes throw an error, https://github.com/SciML/GalacticOptim.jl/blob/270cf27d83fa83a3ded273babe884d3ef014f79a/src/solve.jl#L47 @abhigupta768 add this to the other backends as well.

Reply:
We should actually put that in the readme examples as well to make it explicit